### PR TITLE
Update Vanilla icons to new set

### LINKED ISSUE #3040: Update vanilla icons
Update the icons that are used in Vanilla to use svgs from the new iconset from Brand.

### PR DESCRIPTION
**Pull request by bartaz**
## Done

This updates the icons (icon pattern, and patterns that use those icons).
It doesn't update the external link - that uses separate CSS mask implementation and will be updated in other PR.

Fixes #3040 

## QA

**DEMO SERVICE HAS OLD VERSION, PLEASE QA BRANCH LOCALLY**

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-3083.run.demo.haus/docs/icons)
- Check [icons docs page](https://vanilla-framework-canonical-web-and-design-pr-3083.run.demo.haus/docs/icons)
- Check icon examples:
  - [light theme](https://vanilla-framework-canonical-web-and-design-pr-3083.run.demo.haus/docs/examples/patterns/icons/icons-light)
  - [dark theme](https://vanilla-framework-canonical-web-and-design-pr-3083.run.demo.haus/docs/examples/patterns/icons/icons-dark)
  - [form validation](https://vanilla-framework-canonical-web-and-design-pr-3083.run.demo.haus/docs/examples/patterns/forms/form-validation)

Screenshot 2020-05-21 at 13 54 43

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
Old set:
image

New set:
image

- should we have two question marks in different color? I think just grey should be enough?
- should we be using the filled versions or the outlined versions of error/success/warning? personally I prefer the outlined ersions, they feel more consistent with the rest

File: templates/docs/patterns/icons.md [hunk 1]
```diff
@@ -95,20 +97,16 @@ Our icons have two predefined color styles: light and dark. The light variant is
...
       </div>
-      <div class="p-card col-3 u-vertically-center">
-        <p><i class="p-icon--question" style="margin-right: 1rem;"></i>p-icon--question</p>
-      </div>
     </div>
 
   </div>
 </section>
 
+<span class="p-label--deprecated">Deprecated</span> The `.p-icon--question` has been deprecated and will be removed in future version v3.0. Use an icon with `.p-icon--help` class name instead.
```
Comment (lyubomir-popov) [inline templates/docs/patterns/icons.md:108]:
```suggestion
<span class="p-label--deprecated">Deprecated</span> The `.p-icon--question` class has been deprecated and will be removed in version 3.0. Use class `.p-icon--help` instead.
```

File: templates/docs/patterns/icons.md [hunk 2]
```diff
@@ -95,20 +97,16 @@ Our icons have two predefined color styles: light and dark. The light variant is
...
+
 ### Dark theme
 
 Our dark-themed icons are available when placed within `.p-strip--dark`, icon colors are reverted to ensure legibility as shown in our example.
```
Comment (lyubomir-popov) [inline templates/docs/patterns/icons.md:112]: This should say inverted I believe.

**Comment by lyubomir-popov** [comment]
Adding don't merge until we get a fixed indeterminate icon. The current one has gaps that are not perpendicular to the circle.

**Comment by bartaz** [comment]
I don't mind either way, I guess we are not gonna release that until we have spinner anyway, so we can keep it open.

**Comment by lyubomir-popov** [comment]
Spinner issue mentioned above: 
image

**Pull request merged**